include/bluetooth/services/cts_client.h

```diff
@@ -0,0 +1,152 @@
+	struct bt_cts_current_time_char current_time;
+};
+
+struct bt_cts_client;
+
+/**@brief Current Time Service client event handler type. */
+typedef void (*bt_cts_client_evt_handler)(struct bt_cts_client *cts_c,
+					  const struct bt_cts_client_evt *evt);
+
+/**@brief Current Time Service client structure. This structure contains status information for the client. */
+struct bt_cts_client {
+	/** Connection object. */
+	struct bt_conn *conn;
+
+	/** Handle of the Current Time Characteristic. */
+	uint16_t handle_ct;
+
+	/** Handle of the CCCD of the Current Time Characteristic. */
+	uint16_t handle_ct_ccc;
+
+	/** GATT read parameters. */
+	struct bt_gatt_read_params read_params;
+
+	/** Semaphore for GATT read parameters. */
+	struct k_sem read_params_sem;
+
+	/** Event handler to be called for handling events from the Current Time Service client. */
+	bt_cts_client_evt_handler evt_handler;
+};
+
+/**@brief Function for initializing the Current Time Service client.
+ *
+ * @param[out] cts_c       Current Time Service client structure. This structure must
+ *                         be supplied by the application. It is initialized by this
+ *                         function and can later be used to identify this particular client
+ *                         instance.
+ * @param[in]  evt_handler Event handler for Current Time Service client.
+ *
+ * @retval 0 If the client was initialized successfully.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_cts_client_init(struct bt_cts_client *cts_c,
+		       bt_cts_client_evt_handler evt_handler);
+
+/**@brief Function for clearing the CTS client connection.
+ *
+ * @param[in] cts_c  Current Time Service client structure.
+ */
+void bt_cts_client_on_disconnected(struct bt_cts_client *cts_c);
+
+/**@brief Function for reading the peer's Current Time Service Current Time characteristic.
+ *
+ * @param[in] cts_c    Current Time Service client structure.
+ * @param[in] timeout  Time-out for taking GATT read parameters semaphore.
+ *
+ * @retval 0 If all operations are successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_cts_client_current_time_read(struct bt_cts_client *cts_c,
+				    k_timeout_t timeout);
+
+/**@brief Function for assigning handles to this instance of cts_c.
+ *
+ * @details Call this function when a link has been established with a peer to
+ *          associate the link to this instance of the module. This makes it
+ *          possible to handle several links and associate each link to a particular
+ *          instance of this module.
+ *
+ * @param[in]     dm     Discovery object.
+ * @param[in,out] cts_c  Pointer to the CTS client structure instance for associating the link.
+ *
+ * @retval 0 If the operation is successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_cts_client_handles_assign(struct bt_gatt_dm *dm,
+				 struct bt_cts_client *cts_c);
+
```

Comment:
Should we have an subscribe function as well? with a notify callback.
Could be used for receiving time adjusts (with the adjust reason set).

Reply:
Will do.